### PR TITLE
Verify params[:user] before accessing.

### PR DESCRIPTION
**Pull request by ashercodeorg**
Though this doesn't fix https://app.honeybadger.io/projects/3240/faults/31777337, the hope is that it will mitigate the frequency of the exception. Further investigation planned...

### CODE REVIEW

File: dashboard/app/controllers/registrations_controller.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ def update
     @user = User.find(current_user.id)
 
     # If email has changed for a non-teacher: clear confirmed_at but don't send notification email
-    if params[:user][:email].present? && !@user.confirmation_required?
+    if  !@user.confirmation_required? && params[:user][:email].present?
```
Comment (ashercodeorg) [inline dashboard/app/controllers/registrations_controller.rb]: Note that, based on L26, it seems as if `params[:user]` is expected to be sometimes `nil`.

Comment (aoby) [reply dashboard/app/controllers/registrations_controller.rb]: Then how about `params.try(:[], :user).try(:[], :email).present?` ?

Comment (ashercodeorg) [reply dashboard/app/controllers/registrations_controller.rb]: Done (added `params[:user] &&`). Let me know if you prefer the `try` syntax, I consider it slightly less readable when nested.

Comment (aoby) [reply dashboard/app/controllers/registrations_controller.rb]: No strong preference. 👍

**Pull request merged**